src/lib/Libifl/pbsD_msgjob.c

```diff
@@ -202,6 +202,50 @@ char *extend;
 	if (pbs_client_thread_init_thread_context() != 0)
 		return pbs_errno;
 
+	/* first verify the resource list in keep_select option */
+	if (extend) {
+		struct attrl *attrib = NULL;
+		char ebuff[200], *erp, *emsg = NULL;
+		int i;
+		struct pbs_client_thread_connect_context *con;
+		char nd_ct_selstr[20];
```

Comment:
Why is this array 20 characters?

Reply:
20 is enough to fit string `"select=%d", INT_MAX`